main.go

```
@@ -0,0 +1,85 @@
+package main
+
+import (
+	"log"
+	"net/http"
+	"time"
+
+	"github.com/gorilla/websocket"
+)
+
+type Message struct {
+	Data int `json:"data"`
+}
+
+func writeData(ws *websocket.Conn, quit <-chan struct{}) {
+	data := Message{
+		Data: 0,
+	}
+	for {
+		select {
+		case <-quit:
+			log.Println("received stop signal, stopping write")
+			return
```

In Go, like most languages, `break` is used to exit (escape) the for/switch that you are currently executing.

`return` will exit the entire method you are currently executing (and possibly return a value to the caller, optional).

In your case, you should use `break` since there isn't anything else executing bellow, using `break` and `return` in the same function will lead to errors as the code grows, as a rule of thumb always fail early.